#! /bin/sh -e

# DP: Set DOT_MULTI_TARGET to true.
# DP: Accept additional values known by dot for DOT_IMAGE_FORMAT.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- ./src/configoptions.cpp~	2009-02-08 12:08:33.000000000 +0100
+++ ./src/configoptions.cpp	2009-05-02 15:28:30.000000000 +0200
@@ -1998,8 +1998,21 @@
                  "png"
                 );
   ce->addValue("png");
+  ce->addValue("png:cairo");
+  ce->addValue("png:cairo:cairo");
+  ce->addValue("png:cairo:gd");
+  ce->addValue("png:gd");
+  ce->addValue("png:gd:gd");
   ce->addValue("jpg");
+  ce->addValue("jpg:cairo");
+  ce->addValue("jpg:cairo:gd");
+  ce->addValue("jpg:gd");
+  ce->addValue("jpg:gd:gd");
   ce->addValue("gif");
+  ce->addValue("gif:cairo");
+  ce->addValue("gif:cairo:gd");
+  ce->addValue("gif:gd");
+  ce->addValue("gif:gd:gd");
   ce->addDependency("HAVE_DOT");
   //----
   cs = cfg->addString(
@@ -2062,7 +2075,7 @@
                  "files in one run (i.e. multiple -o and -T options on the command line). This \n"
                  "makes dot run faster, but since only newer versions of dot (>1.8.10) \n"
                  "support this, this feature is disabled by default.",
-                 FALSE
+                 TRUE
                 );
   cb->addDependency("HAVE_DOT");
   //----
